CMakeLists.txt

```diff
@@ -123,14 +130,10 @@ ENDIF()
 
 # Add OpenMP
 IF(BUILD_WITH_MULTITHREADS)
-  FIND_PACKAGE(OpenMP)
+  FIND_PACKAGE(OpenMP REQUIRED)
 ENDIF()
-IF(OPENMP_FOUND AND BUILD_WITH_MULTITHREADS)
-  IF(APPLE)
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xpreprocessor -fopenmp")
-  ELSE()
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
-  ENDIF(APPLE)
+IF(BUILD_WITH_MULTITHREADS)
+  ADD_COMPILE_OPTIONS(${PROJECT_NAME} ${OpenMP_CXX_FLAGS})
```

Comment:
The cmake submodule is defining `-Xpreprocessor` for all the cases. I know this flag is needed for OSX but not for Linux.

Have been tested this compilation flag in Ubuntu and Windows?
If this works, what is the effect in Ubuntu and Windows?